File: app/src/main/java/com/rezolve/gistscanner/data/GistRepository.java
package com.rezolve.gistscanner.data;

import android.support.annotation.NonNull;

import com.rezolve.gistscanner.Remote;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class GistRepository {
    private final IGistDataSource gistApi;

    @Inject
    public GistRepository(@Remote IGistDataSource gistApi) {
        this.gistApi = gistApi;
    }

    public void fetchGistCommentList(@NonNull String gistID, @NonNull String username, @NonNull String password,
                                     @NonNull Callback<GistCommentListResponse> callback) {
        gistApi.getGistCommentList(gistID, username, password, callback);
    }

    public void createdGistComment(@NonNull String gistID, @NonNull String username,
                                   @NonNull String password, @NonNull String comment,
                                   @NonNull Callback<CreateGistResponse> callback) {
        gistApi.createGistComment(gistID, username, password, comment, callback);
    }
}

File: app/src/main/java/com/rezolve/gistscanner/data/GistRepository.kt
package com.rezolve.gistscanner.data

import com.rezolve.gistscanner.model.GistComment
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GistRepository @Inject constructor(private val gistApi: RetrofitGistDataSource) {
    fun fetchGistCommentList(gistID: String, username: String, password: String,
                             callback: NetworkCallback<List<GistComment>>) =
            gistApi.getGistCommentList(gistID, username, password, callback)

    fun createdGistComment(gistID: String, username: String, password: String,
                           comment: String,
                           callback: NetworkCallback<GistComment>) =
            gistApi.createGistComment(gistID, username, password, comment,
                    callback)
}